mypy/util.py

```diff
@@ -411,9 +410,11 @@ def replace_object_state(
             pass
 
 
-def is_sub_path(path1: str, path2: str) -> bool:
-    """Given two paths, return if path1 is a sub-path of path2."""
-    return pathlib.Path(path2) in pathlib.Path(path1).parents
+def is_sub_path(path: str, dir: str) -> bool:
+    """Given two paths, return if path is a sub-path of dir."""
+    if not dir.endswith(os.sep):
+        dir += os.sep
+    return path.startswith(dir)
```

Comment:
Does this will work reliably on Windows, where path separators can be `/` or `\`, and these are largely equivalent? Should we normalize separators on Windows, or can we assume everything is normalized here? I noticed that you added a call to `os.path.normcase`, so maybe this is already ok. 

Maybe update docstring to mention Windows-specific issues, and case insensitivity issues?

Reply:
Documented the shortcomings of the faster function + invariants we require in modulefinder. Also renamed the function so it sounds a little scarier